lib/travis/logs.rb

```diff
@@ -1,11 +1,16 @@
-require 'travis/logs/config'
-
 module Travis
   def self.config
     Logs.config
   end
 
   module Logs
+    autoload :Aggregate, 'travis/logs/aggregate'
+    autoload :App, 'travis/logs/app'
+    autoload :Config, 'travis/logs/config'
+    autoload :Existence, 'travis/logs/existence'
+    autoload :Receive, 'travis/logs/receive'
+    autoload :Sidekiq, 'travis/logs/sidekiq'
```

Comment:
I'm curios: Why the switch to autoload over explicit requires? I've noticed that we are making the opposite trend elsewhere (or at least I think we were back when I last was mostly involved in the Ruby code, which admittedly is a while ago).